catalog/rocks/entry_value_iterator_test.go

```diff
@@ -0,0 +1,41 @@
+package rocks
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/treeverse/lakefs/graveler"
+)
+
+func TestNewEntryToValueIterator(t *testing.T) {
+	var expectedRecords []*graveler.ValueRecord
+	var entryRecords []*EntryRecord
+
+	// generate data
+	for i := 0; i < 3; i++ {
+		record := &EntryRecord{
+			Path:  Path(fmt.Sprintf("path%d", i)),
+			Entry: &Entry{Address: fmt.Sprintf("addr%d", i)},
+		}
+		entryRecords = append(entryRecords, record)
+		expectedRecords = append(expectedRecords, &graveler.ValueRecord{
+			Key:   []byte(record.Path.String()),
+			Value: MustEntryToValue(record.Entry),
+		})
+	}
+
+	// collect the above using entry to value iterator
+	entryIterator := NewFakeEntryIterator(entryRecords)
+	it := NewEntryToValueIterator(entryIterator)
+	defer it.Close()
+
+	var values []*graveler.ValueRecord
+	for it.Next() {
+		values = append(values, it.Value())
+	}
+	// verify that what we produced is what we got from the iterator
+	if diff := deep.Equal(values, expectedRecords); diff != nil {
```

Comment:
check that the iterator ended without an error
```go
if err := it.Err(); err != nil {
    t.Fatal("Iterator ended with error", err)
}
```

Reply:
Done.